src/analysis/seminorm.lean

```diff
@@ -334,11 +375,28 @@ lemma mem_ball_zero : y ∈ ball p 0 r ↔ p y < r := by rw [mem_ball, sub_zero]
 
 lemma ball_zero_eq : ball p 0 r = { y : E | p y < r } := set.ext $ λ x, p.mem_ball_zero
 
+@[simp] lemma ball_zero' (x : E) (hr : 0 < r) : ball (0 : seminorm 𝕜 E) x r = set.univ :=
+begin
+  rw [set.eq_univ_iff_forall, ball],
+  simp [hr],
+end
+
+lemma ball_sup (p : seminorm 𝕜 E) (q : seminorm 𝕜 E) (e : E) {r : ℝ} (hr : 0 < r) :
+  ball (p ⊔ q) e r = ball p e r ∩ ball q e r :=
+begin
+  lift r to nnreal using hr.le,
+  simp_rw [ball, ←set.set_of_and],
+  simp,
+end
```

Comment:
You don't need `r` to be non-zero here:
```suggestion
lemma ball_sup (p : seminorm 𝕜 E) (q : seminorm 𝕜 E) (e : E) {r : ℝ} (hr : 0 ≤ r) :
  ball (p ⊔ q) e r = ball p e r ∩ ball q e r :=
begin
  lift r to nnreal using hr,
  simp_rw [ball, ←set.set_of_and],
  simp,
end
```

Reply:
In fact, it seems you don't even need `hr` at all.

Reply:
```
lemma ball_sup (p : seminorm 𝕜 E) (q : seminorm 𝕜 E) (e : E) {r : ℝ} :
  ball (p ⊔ q) e r = ball p e r ∩ ball q e r :=
by simp only [ball, ←set.set_of_and, pi.sup_apply, sup_lt_iff, coe_sup]
```

Reply:
@mcdoll: see my later comment, my original patch was still stronger than needed.